app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -71,8 +80,9 @@ protected void onCreate(Bundle savedInstanceState) {
         downloadProgress = (ProgressBar) findViewById(R.id.progress);
         downloadProgress.setVisibility(View.VISIBLE);
         downloadProgress.setIndeterminate(true);
-        DataDownload download = new DataDownload();
-        download.downloadVersions();
+        registerReceiver(new NetworkDialogHandler(), new IntentFilter(Context.CONNECTIVITY_SERVICE));
```

Comment:
registerReceiver is already in Mainactivity#onCreate(). Please check.

Or do you mean I should keep it in OpenEventApp#oncreate()? OpenEventApp extends Application.